tests/unit/Hydrators/FamilyTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Choredo\Test\Hydrators;
+
+use Assert\LazyAssertionException;
+use Choredo\Entities;
+use Choredo\Hydrators;
+use PHPUnit\Framework\TestCase;
+use const Choredo\SHORT_DATA_FIELD_MAX_SIZE;
+
+class FamilyTest extends TestCase
+{
+    protected $dataStub = [
+        'name'            => 'test family',
+        'paymentStrategy' => Entities\Family::PAYMENT_STRATEGY_PER_CHILD,
+        'weekStartDay'    => 'sunday',
+    ];
+
+    public function testHydratorGeneratesIdWhenNotProvided()
+    {
+        $data = $this->dataStub;
+
+        $hydrator = new Hydrators\Family();
+        $family = $hydrator->hydrate($data);
+
+        $this->assertInstanceOf(Entities\Family::class, $family);
+        $this->assertNotEmpty($family->getId());
+    }
+
+    /**
+     * @expectedException \Assert\LazyAssertionException
+     * @expectedExceptionMessage Family::id: Value "this_is_not_a_uuid" is not a valid UUID.
+     */
+    public function testHydratorThrowsExceptionOnInvalidId()
+    {
+        $data = array_merge($this->dataStub, ['id' => 'this_is_not_a_uuid']);
+        $hydrator = new Hydrators\Family();
+        $hydrator->hydrate($data);
+    }
+
+    /**
+     * @param $completionThreshold
+     * @param $exceptionMessage
+     *
+     * @dataProvider invalidPaymentStrategyCompletionThresholdCombinationsProvider
+     */
+    public function testHydratorHandlesValidationOfPerChorePaymentStrategy($completionThreshold, $exceptionMessage)
+    {
+        $data = array_merge(
+            $this->dataStub,
+            [
+                'paymentStrategy'     => Entities\Family::PAYMENT_STRATEGY_PER_CHORE,
+                'completionThreshold' => $completionThreshold,
+            ]
+        );
+
+        $this->expectException(LazyAssertionException::class);
+        $this->expectExceptionMessage($exceptionMessage);
+        (new Hydrators\Family())->hydrate($data);
+    }
+
+    public function invalidPaymentStrategyCompletionThresholdCombinationsProvider(): array
+    {
+        return [
+            [null, "Family::completionThreshold: Value \"<NULL>\" is empty, but non empty value was expected."],
+            [0, "Family::completionThreshold: Value \"0\" is empty, but non empty value was expected."],
```

Comment:
MIN_COMPLETION_THRESHOLD is 0... so shouldn't 0 be a valid value?

Reply:
Yep, good catch. It got picked up by a notEmpty() assertion which should have been isNull() and I didn't think it through.